src/adapters/kafka/index.ts

```diff
@@ -42,18 +40,29 @@ class KafkaAdapter extends Adapter {
     await consumer.subscribe({ topics: subscribedChannels, fromBeginning: true })
     await consumer.run({
       eachMessage: async ({ topic, partition, message }) => {
-        const msg = this._createMessage(topic, message)
+        const msg = this._createMessage(topic, partition, message)
         this.emit('message', msg, consumer)
       },
-    }) 
+    })
+    
+    const producer = kafka.producer()
+    await producer.connect()
+    await producer.send({
+      topic: 'glee-topic',
+      messages: [{ value: 'Hello, Glee!' }],
+    })
+  }
+
+  async send(message: GleeMessage) {
+    
```

Comment:
```suggestion
    const producer = kafka.producer()
    await producer.connect()
    await producer.send({
      topic: message.channel,
      messages: [{
        value: message.payload,
      }],
    })
```

You can pass more details to the message object above. The structure documentation is here: https://kafka.js.org/docs/producing#message-structure.

In `_createMessage`, you converted a Kafka message to a GleeMessage. Here you have to do the opposite. You get a GleeMessage and have to convert it to a Kafka message.

You probably have to call `await producer.disconnect()` after sending the message.